integration-tests/src/test/java/org/apache/fineract/integrationtests/LoanAccountPaymentAllocationWithOverlappingDownPaymentInstallmentTest.java

```diff
@@ -637,6 +634,325 @@ public void loanAccountWithEnableDownPaymentAndDisableAutoRepaymentForDownPaymen
+
+            // make another repayment and verify downpayment installment gets paid off on 3rd april
+            final PostLoansLoanIdTransactionsResponse repaymentTransaction_4 = loanTransactionHelper.makeLoanRepayment(loanExternalIdStr,
+                    new PostLoansLoanIdTransactionsRequest().dateFormat("dd MMMM yyyy").transactionDate("3 April 2023").locale("en")
+                            .transactionAmount(100.00));
+
+            loanDetails = loanTransactionHelper.getLoanDetails(loanId.longValue());
+
+            assertNotNull(loanDetails.getRepaymentSchedule());
+
+            // verify down payment installment gets paid off
+
+            // first period [2023-03-03 to 2023-03-03] first down payment installment
+            assertEquals(1, loanDetails.getRepaymentSchedule().getPeriods().get(1).getPeriod());
+            assertEquals(LocalDate.of(2023, 3, 3), loanDetails.getRepaymentSchedule().getPeriods().get(1).getFromDate());
+            assertEquals(LocalDate.of(2023, 3, 3), loanDetails.getRepaymentSchedule().getPeriods().get(1).getDueDate());
+            assertEquals(50.00, loanDetails.getRepaymentSchedule().getPeriods().get(1).getTotalInstallmentAmountForPeriod());
+            assertEquals(50.00, loanDetails.getRepaymentSchedule().getPeriods().get(1).getTotalPaidForPeriod());
+            assertEquals(true, loanDetails.getRepaymentSchedule().getPeriods().get(1).getComplete());
+
+            // second period [2023-03-03 to 2023-04-03] regular installment, principal gets paid first
+            assertEquals(2, loanDetails.getRepaymentSchedule().getPeriods().get(4).getPeriod());
+            assertEquals(LocalDate.of(2023, 3, 3), loanDetails.getRepaymentSchedule().getPeriods().get(4).getFromDate());
+            assertEquals(LocalDate.of(2023, 4, 3), loanDetails.getRepaymentSchedule().getPeriods().get(4).getDueDate());
+            assertEquals(225.0, loanDetails.getRepaymentSchedule().getPeriods().get(4).getTotalInstallmentAmountForPeriod());
+            assertEquals(225.0, loanDetails.getRepaymentSchedule().getPeriods().get(4).getTotalPaidForPeriod());
+            assertEquals(225.0, loanDetails.getRepaymentSchedule().getPeriods().get(4).getPrincipalPaid());
+            assertEquals(0.0, loanDetails.getRepaymentSchedule().getPeriods().get(4).getTotalOutstandingForPeriod());
+            assertEquals(0.0, loanDetails.getRepaymentSchedule().getPeriods().get(4).getFeeChargesDue());
+            assertEquals(0.0, loanDetails.getRepaymentSchedule().getPeriods().get(4).getFeeChargesPaid());
+            assertEquals(0.0, loanDetails.getRepaymentSchedule().getPeriods().get(4).getPenaltyChargesDue());
+            assertEquals(0.0, loanDetails.getRepaymentSchedule().getPeriods().get(4).getPenaltyChargesPaid());
+            assertEquals(true, loanDetails.getRepaymentSchedule().getPeriods().get(4).getComplete());
+
+            // third period [2023-04-03 to 2023-04-03] overlapping down payment installment
```

Comment:
Just for double checking. We don't want to create two separate entries for the two downpayments happend on the same day, right?